src/services/posts/fetch-posts.test.js

```diff
@@ -0,0 +1,84 @@
+const {expect} = require('chai')
+const db = require("../../db")
+const fetchPosts = require("./fetch-posts")
+const {fixtures, resetDBTable} = require('../../../test/utils')
+
+describe('Fetch all posts', () => {
+
+	const numberOfPosts = 5
+	let user 
+	let posts
+
+	before(async () => {
+        
+		user = await fixtures.insertUser()
+		await resetDBTable('posts')
+		posts = await fixtures.insertMultiplePosts(user.id, numberOfPosts)
+
+	 })
+	 
+	it('should fetch posts', async () =>{
+		const actualPosts = await fetchPosts()
+
+		const result = await db.query(`
+		SELECT posts.id, posts."userId" , posts.title, posts.image,
+        posts.content,posts.published, posts."createdAt", posts.type,
+        posts."likesCount",
+        users."firstName", users."lastName", users.email, 
+        users."profilePictureUrl" 
+	    FROM posts
+	    INNER JOIN users on posts."userId" = users.id
+		ORDER BY posts."createdAt" DESC;
+        `)
+		return expect(actualPosts).to.eql(result.rows)	
+	})
+
+	it('should fetch flagged posts', async () =>{
+	
+		await fixtures.insertPostFlag({
+			userId : user.id ,
+			postId : posts[0].id
+		})
+		const actualPosts = await fetchPosts(true)
+        
+
+		const result = await db.query(`
+		SELECT posts.id, posts."userId" , posts.title, posts.image,
+		posts.content,posts.published, posts."createdAt", posts.type,
+		posts."likesCount", posts."flagsCount",
+		users."firstName", users."lastName", users.email,
+		users."profilePictureUrl" 
+		FROM posts
+		INNER JOIN users on posts."userId" = users.id
+		WHERE posts."flagsCount" > 0
+		ORDER BY "flagsCount" DESC;
+        `)
+
+		return expect(actualPosts).to.eql(result.rows)	
+	})
+
+	it('should fetch unflagged posts', async () =>{
+		await resetDBTable('post_flags')
+	
+		await fixtures.insertPostFlag({
+			userId : user.id ,
+			postId : posts[0].id
+		})
+		const actualPosts = await fetchPosts(false)
+        
+
+		const result = await db.query(`
+		SELECT posts.id, posts."userId" , posts.title, posts.image,
+		posts.content,posts.published, posts."createdAt", posts.type,
+		posts."likesCount", posts."flagsCount",
+		users."firstName", users."lastName", users.email,
+		users."profilePictureUrl" 
+		FROM posts
+		INNER JOIN users on posts."userId" = users.id
+ 		WHERE posts."flagsCount" = 0
+		ORDER BY "flagsCount" DESC;
+        `)
+
+		return expect(actualPosts).to.eql(result.rows)	
```

Comment:
in line with what we're confirming based on the test case description, `should fetch unflagged posts`, we should be confirming that the posts returned by the service are really unflagged by checking the flagsCount

else we're not confirming much since we're using the same query being used in the service. if query is wrong, bug goes unnoticed because tests will still pass since we're basically equating results from the same query

```suggestion
		expect(actualPosts).to.eql(result.rows)
	        actualPosts.map(post => expect(post.flagsCount).to.equal(0))

```

Reply:
yes please